### PR TITLE
[CLN] website_sale: models cleanup

### PR DESCRIPTION
**Pull request by Feyensv**
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Julien00859 — APPROVED** [review]

File: addons/website_sale/models/res_config_settings.py [hunk 1]
```diff
@@ -1,40 +1,76 @@
...
+    )
+    send_abandoned_cart_email = fields.Boolean(
+        string="Abandoned Email",
+        related='website_id.send_abandoned_cart_email',
+        readonly=False,
+    )
+    salesperson_id = fields.Many2one(
+        related='website_id.salesperson_id',
+        readonly=False,
+    )
```
Comment (Demesmaeker) [inline addons/website_sale/models/res_config_settings.py:53]: We lost the domain, is it intentional?

File: addons/website_sale/models/res_config_settings.py [hunk 2]
```diff
@@ -1,40 +1,76 @@
...
-    cart_recovery_mail_template = fields.Many2one('mail.template', string='Cart Recovery Email', domain="[('model', '=', 'sale.order')]",
-                                                  related='website_id.cart_recovery_mail_template_id', readonly=False)
-    cart_abandoned_delay = fields.Float(string="Send After", related='website_id.cart_abandoned_delay', readonly=False)
-    send_abandoned_cart_email = fields.Boolean('Abandoned Email', related='website_id.send_abandoned_cart_email', readonly=False)
+    # Website-dependent settings
     add_to_cart_action = fields.Selection(related='website_id.add_to_cart_action', readonly=False)
+    cart_recovery_mail_template = fields.Many2one(
+        related='website_id.cart_recovery_mail_template_id',
+        readonly=False,
+    )
```
Comment (Demesmaeker) [inline addons/website_sale/models/res_config_settings.py:40]: same

File: addons/website_sale/models/res_config_settings.py [hunk 3]
```diff
@@ -45,32 +81,17 @@ class ResConfigSettings(models.TransientModel):
...
 
-    @api.model
-    def get_values(self):
-        res = super(ResConfigSettings, self).get_values()
-        res.update(
-            enabled_extra_checkout_step=self.env.ref('website_sale.extra_info').active,
-            enabled_buy_now_button=self.env.ref('website_sale.product_buy_now').active,
-        )
-        return res
+    # TODO move to sale (and move group_discount_per_so_line from product to sale as well)
```
Comment (Demesmaeker) [inline addons/website_sale/models/res_config_settings.py:87]: :eyes:

File: addons/website_sale/models/res_config_settings.py [hunk 4]
```diff
@@ -1,40 +1,76 @@
...
+        readonly=False,
+    )
 
-    module_delivery_mondialrelay = fields.Boolean("Mondial Relay Connector")
-    group_product_pricelist = fields.Boolean(
-        compute='_compute_group_product_pricelist', store=True, readonly=False)
-
-    enabled_extra_checkout_step = fields.Boolean(string="Extra Step During Checkout")
-    enabled_buy_now_button = fields.Boolean(string="Buy Now")
+    # Additional settings
```
Comment (Demesmaeker) [inline addons/website_sale/models/res_config_settings.py:74]: Line or not line after the comment announcing the group of fields? That's the question ^^

File: addons/website_sale/models/res_config_settings.py [hunk 5]
```diff
@@ -1,40 +1,76 @@
...
+    salesteam_id = fields.Many2one(related='website_id.salesteam_id', readonly=False)
+    website_sale_prevent_zero_price_sale = fields.Boolean(
+        string="Prevent Sale of Zero Priced Product",
+        related='website_id.prevent_zero_price_sale',
+        readonly=False,
+    )
+    website_sale_contact_us_button_url = fields.Char(
+        related='website_id.contact_us_button_url',
+        readonly=False,
+    )
```
Comment (Demesmaeker) [inline addons/website_sale/models/res_config_settings.py:63]: String now missing, might still be useful? (didn't verify)

File: addons/website_sale/models/res_config_settings.py [hunk 6]
```diff
@@ -1,40 +1,76 @@
...
+    website_sale_enabled_portal_reorder_button = fields.Boolean(
+        related='website_id.enabled_portal_reorder_button',
+        readonly=False,
+    )
```
Comment (Demesmaeker) [inline addons/website_sale/models/res_config_settings.py:67]: same

File: addons/website_sale/models/res_config_settings.py [hunk 7]
```diff
@@ -94,28 +115,51 @@ def _inverse_account_on_checkout(self):
...
+            extra_step_view.active = self.enabled_extra_checkout_step
+        buy_now_view = self.env.ref('website_sale.product_buy_now')
+        if buy_now_view.active != self.enabled_buy_now_button:
+            buy_now_view.active = self.enabled_buy_now_button
+
+    #=== ACTION METHODS ===#
+
+    def action_open_abandoned_cart_mail_template(self):
         return {
             'name': _('Customize Email Templates'),
```
Comment (Demesmaeker) [inline addons/website_sale/models/res_config_settings.py:142]:
```suggestion
            'name': _("Customize Email Templates"),
```

File: addons/website_sale/models/res_partner.py [hunk 8]
```diff
@@ -8,7 +8,11 @@
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    last_website_so_id = fields.Many2one('sale.order', compute='_compute_last_website_so_id', string='Last Online Sales Order')
+    last_website_so_id = fields.Many2one(
+        comodel_name='sale.order',
+        compute='_compute_last_website_so_id',
+        string="Last Online Sales Order",
+    )
```
Comment (Demesmaeker) [inline addons/website_sale/models/res_partner.py:15]:
```suggestion
    last_website_so_id = fields.Many2one(
        comodel_name='sale.order',
        string="Last Online Sales Order",
        compute='_compute_last_website_so_id',
    )
```
We could be more coherent with the order in our field's declaration IMHO, you sometime changed the order, but not always

File: addons/website_sale/models/sale_order.py [hunk 9]
```diff
@@ -13,69 +13,41 @@
 
 
 class SaleOrder(models.Model):
-    _inherit = "sale.order"
+    _inherit = 'sale.order'
+
+    website_id = fields.Many2one(
+        comodel_name='website',
+        string="Website",
```
Comment (Demesmaeker) [inline addons/website_sale/models/sale_order.py:20]: Is that string necessary?

File: addons/website_sale/models/sale_order.py [hunk 10]
```diff
@@ -127,6 +105,142 @@ def _search_abandoned_cart(self, operator, value):
...
+                    'parent_id': parent_id,
+                    'type': 'delivery',
+                    'name': name,
+                    'street': street,
+                    'city': city,
+                    'state_id': state,
+                    'zip': zip_code,
+                    'country_id': country,
+                    'email': email,
+                    'phone': phone
```
Comment (Demesmaeker) [inline addons/website_sale/models/sale_order.py:189]:
```suggestion
                    'phone': phone,
```

File: addons/website_sale/models/sale_order.py [hunk 11]
```diff
@@ -127,6 +105,142 @@ def _search_abandoned_cart(self, operator, value):
...
+            email = order.partner_shipping_id.email
+            phone = order.partner_shipping_id.phone
+
+            # we can check if the current partner has a partner of type "delivery" that has the same address
+            existing_partner = order.env['res.partner'].search([
+                ('street', '=', street),
+                ('city', '=', city),
+                ('state_id', '=', state),
+                ('country_id', '=', country),
+                ('type', '=', 'delivery')
```
Comment (Demesmaeker) [inline addons/website_sale/models/sale_order.py:172]:
```suggestion
                ('type', '=', 'delivery'),
```

File: addons/website_sale/models/sale_order.py [hunk 12]
```diff
@@ -127,6 +105,142 @@ def _search_abandoned_cart(self, operator, value):
...
+            # retrieve all the data :
+            # name, street, city, state, zip, country
+            name = order.partner_shipping_id.name
+            street = order_location['pick_up_point_address']
+            city = order_location['pick_up_point_town']
+            zip_code = order_location['pick_up_point_postal_code']
+            country = order.env['res.country'].search([('code', '=', order_location['pick_up_point_country'])]).id
+            state = order.env['res.country.state'].search([
+                ('code', '=', order_location['pick_up_point_state']),
+                ('country_id.id', '=', country)
```
Comment (Demesmaeker) [inline addons/website_sale/models/sale_order.py:160]:
```suggestion
                ('country_id.id', '=', country),
```

File: addons/website_sale/models/website.py [hunk 13]
```diff
@@ -11,106 +10,146 @@
...
+    pricelist_ids = fields.One2many(
+        comodel_name='product.pricelist',
+        compute="_compute_pricelist_ids",
+        string="Price list available for this Ecommerce/Website",
+    )
+    # Technical: Used to recompute pricelist_ids
+    all_pricelist_ids = fields.One2many(
+        comodel_name='product.pricelist',
+        inverse_name='website_id',
+        string="All pricelists",
```
Comment (Demesmaeker) [inline addons/website_sale/models/website.py:149]: useless string then

File: addons/website_sale/models/website.py [hunk 14]
```diff
@@ -11,106 +10,146 @@
...
-            ('list_price asc', _('Price - Low to High')),
-            ('list_price desc', _('Price - High to Low')),
-        ]
     shop_default_sort = fields.Selection(selection='_get_product_sort_mapping', default='website_sequence asc', required=True)
 
-    shop_extra_field_ids = fields.One2many('website.sale.extra.field', 'website_id', string='E-Commerce Extra Fields')
+    shop_extra_field_ids = fields.One2many(
+        comodel_name='website.sale.extra.field',
+        inverse_name='website_id',
+        string="E-Commerce Extra Fields",
```
Comment (AntoineVDV) [inline addons/website_sale/models/website.py:84]: move up

File: addons/website_sale/models/website.py [hunk 15]
```diff
@@ -11,106 +10,146 @@
...
-            ('stay', 'Stay on Product Page'),
-            ('go_to_cart', 'Go to cart'),
+            ('carousel', "Carousel"),
+            ('grid', "Grid"),
         ],
-        default='stay')
-    auth_signup_uninvited = fields.Selection(default='b2c')
-    account_on_checkout = fields.Selection(
-        string="Customer Accounts",
+        default='carousel',
```
Comment (AntoineVDV) [inline addons/website_sale/models/website.py:92]: move down

**Review by Demesmaeker — COMMENTED** [review]
:duck: 

Nice!! =D

File: addons/website_sale/models/website.py [hunk 16]
```diff
@@ -11,106 +10,146 @@
...
-        ], default='carousel', required=True,
-    )
-    product_page_grid_columns = fields.Integer(default=2)
-    product_page_image_width = fields.Selection([
-        ('none', 'Hidden'),
-        ('50_pc', '50 %'),
-        ('66_pc', '66 %'),
-        ('100_pc', '100 %'),
-        ], default='50_pc', required=True,
+        default='50_pc',
```
Comment (AntoineVDV) [inline addons/website_sale/models/website.py:102]: move down

File: addons/website_sale/models/website.py [hunk 17]
```diff
@@ -11,106 +10,146 @@
...
-        ('big', 'Big'),
-        ], default='small', required=True,
+    product_page_image_spacing = fields.Selection(
+        selection=[
+            ('none', "None"),
+            ('small', "Small"),
+            ('medium', "Medium"),
+            ('big', "Big"),
+        ],
+        default='small',
```
Comment (AntoineVDV) [inline addons/website_sale/models/website.py:112]: move down

File: addons/website_sale/models/website.py [hunk 18]
```diff
@@ -11,106 +10,146 @@
...
+    # Computed fields
     enabled_delivery = fields.Boolean(string="Enable Shipping", compute='_compute_enabled_delivery')
+    fiscal_position_id = fields.Many2one(
+        comodel_name='account.fiscal.position',
+        compute='_compute_fiscal_position_id',
+    )
+    pricelist_id = fields.Many2one(
+        comodel_name='product.pricelist',
+        compute='_compute_pricelist_id',
+        string="Default Pricelist if any",
```
Comment (AntoineVDV) [inline addons/website_sale/models/website.py:133]: move up

File: addons/website_sale/models/website.py [hunk 19]
```diff
@@ -11,106 +10,146 @@
...
+    )
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+        compute='_compute_currency_id',
+        string="Default Currency",
+    )
+    pricelist_ids = fields.One2many(
+        comodel_name='product.pricelist',
+        compute="_compute_pricelist_ids",
+        string="Price list available for this Ecommerce/Website",
```
Comment (AntoineVDV) [inline addons/website_sale/models/website.py:143]: move up

File: addons/website_sale/models/website.py (hunk 13 above)
Comment (AntoineVDV) [inline addons/website_sale/models/website.py:149]: move up

File: addons/website_sale/models/website_snippet_filter.py
Comment (AntoineVDV) [inline addons/website_sale/models/website_snippet_filter.py:1]: clean  `vproduct_cross_selling`

File: addons/website_sale/models/website_visitor.py
Comment (AntoineVDV) [inline addons/website_sale/models/website_visitor.py:1]: clean all fields

File: addons/website_sale/models/res_config_settings.py (hunk 1 above)
Comment (Feyensv) [reply addons/website_sale/models/res_config_settings.py:53]: Nice catch

File: addons/website_sale/models/res_config_settings.py (hunk 2 above)
Comment (Feyensv) [reply addons/website_sale/models/res_config_settings.py:40]: inherited through the related

File: addons/website_sale/models/res_config_settings.py (hunk 6 above)
Comment (Feyensv) [reply addons/website_sale/models/res_config_settings.py:68]: inherited from related

File: addons/website_sale/models/res_partner.py (hunk 8 above)
Comment (Feyensv) [reply addons/website_sale/models/res_partner.py]: We decided on an order with ANV, now defined in our team guidelines.

**Comment by AntoineVDV** [comment]
@robodoo override=ci/security
Reason: `sanitize`: nothing new; fields are just reordered
Link to ci/security build (master): https://runbot.odoo.com/runbot/build/57565123

**Comment by Feyensv** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**